alternator/streams.cc

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright 2019 ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <type_traits>
+#include <boost/lexical_cast.hpp>
+
+#include "base64.hh"
+#include "log.hh"
+#include "database.hh"
+
+#include "cdc/log.hh"
+#include "cdc/generation.hh"
+#include "db/system_distributed_keyspace.hh"
+#include "utils/UUID_gen.hh"
+
+#include "executor.hh"
+#include "tags_extension.hh"
+#include "rmw_operation.hh"
+
+template<typename ValueType, typename T>
+struct from_string_helper {
+    static bool Is(const ValueType& v) {
+        try {
+            Get(v);
+            return true;
+        } catch (...) {
+            return false;
+        }
+    }
+    static T Get(const ValueType& v) {
+        auto s = rjson::to_string_view(v);
+        if constexpr (std::is_constructible_v<T, std::string_view>) {
+            return T(s);
+        } else if constexpr (std::is_constructible_v<T, sstring>) {
+            return T(sstring(s));
+        } else {
+            return boost::lexical_cast<T>(s.data(), s.size());
+        }
+    }
+    static ValueType& Set(ValueType&, T) = delete;
```

Comment:
The type conversion template is expected to convert data for both data set with and without allocator. However, because we have a slightly more narrow use case, we limit it to the case _with_ allocator, i.e. rjson.hh:184. Otherwise we would have to produce values that are transient and do not need delete, which we probably cannot. 
Deleting the function will ensure that compiler will complain early if anyone tried to use the converter the wrong way.